src/manager/monitor_proxy.c

```diff
@@ -0,0 +1,262 @@
+#include "libhirte/log/log.h"
+
+#include "job.h"
+#include "manager.h"
+#include "monitor.h"
+#include "monitor_proxy.h"
+#include "node.h"
+
+Subscription *subscription_proxy_new(ProxyMonitor *monitor, const char *node, const char *unit) {
+        _cleanup_subscription_ Subscription *subscription = malloc0(sizeof(Subscription));
+        if (subscription == NULL) {
+                return NULL;
+        }
+
+        subscription->ref_count = 1;
+        subscription->monitor = proxy_monitor_ref(monitor);
+        subscription->free_monitor = (free_func_t) proxy_monitor_unref;
+        LIST_INIT(subscriptions, subscription);
+        LIST_INIT(all_subscriptions, subscription);
```

Comment:
A bunch of this should be shared in a common function rather than duplicated like this.

Reply:
Moved to `subscription_new` and renamed the existing functions to `create_<monitor>_subscription`.